sdks/python/apache_beam/runners/worker/bundle_processor.py

```diff
@@ -114,17 +114,26 @@ def __init__(self, operation_name, step_name, consumers, counter_factory,
         operations.ConsumerSet(self.counter_factory, self.step_name, 0,
                                consumers.itervalues().next(),
                                self.windowed_coder)]
+    self.input_count = 0
 
   def process(self, windowed_value):
+    self.input_count += 1
     self.output(windowed_value)
 
   def process_encoded(self, encoded_windowed_values):
     input_stream = coder_impl.create_InputStream(encoded_windowed_values)
     while input_stream.size() > 0:
+      self.input_count += 1
       decoded_value = self.windowed_coder.get_impl().decode_from_stream(
           input_stream, True)
       self.output(decoded_value)
 
+  def progress_metrics(self):
+    metrics = super(DataInputOperation, self).progress_metrics()
+    metrics.processed_elements.measured.input_element_counts[
+        'GRPC_INPUT'] = self.input_count
```

Comment:
I'm not sure whether we should be recording this.

Reply:
See the comment above.

Reply:
As a future idea, what do you think of defining a proto message that encodes all the constants for well known cross language URNs and other things like this GRPC_INPUT as a proto message that can be read so we can share the constants across multiple languages without duplicating the definition arbitrarily?

Reply:
Yes, didn't we look into this before and determine there's not a good way to store arbitrary strings in proto messages? 

The next best thing seems to be depending on some shared text file with lines like URN="beam:apache:..." Of course we'd still have to hard code the lhs names, but it'd be an error on failed lookup and we'd force consistency of values (and perhaps we could add some auto-generation to java, go, etc. files that would expose these as constants).